### ISSUE TITLE
Error: "Cant find [Assembly].deps.json" (but it exists)

### ISSUE THREAD

**Issue by SeanKilleen** [aws/aws-lambda-dotnet#854]
Hi all,

Newer to AWS & Lambdas but I think I may have come across a problem. Hoping it's something I'm doing.

## Setup / Versions

* I have an API gateway that proxies to a Lambda whose code is storied in an S3 bucket for the code -- api gateway, lambda, and bucket deployed via terraform 
* The API gateway references a code.zip file in the S3 bucket.
* I am using the dotnet global tool "amazon.lambda.tools" v5.1.0
* I am using aws cli v2.1.29

## Reproduction Steps

* Created a new web API project: `dotnet new serverless.AspNetCoreWebAPI`
* Updated the handler function in the API gateway to match that specified in the project that is created above
* Run `dotnet restore`, `dotnet build`
* Run `dotnet lambda package code.zip`
* Verified that `[AssemblyName].deps.json` exists at the root of the zip file.
* Upload the zip file to the S3 bucket that feeds the API gateway lambda
* Test the API via the management console

**Expected behavior**: see the default response for the `/api/values` route
**Actual behavior**: See an error: `"Could not find the required 'LbsApi.deps.json'.  This file should be present at the root of the deployment package."` (this file is in the root of the zip file).

### Environment

* Build Version: Amazon.Lambda.AspNetCoreServer v5.3.1
* Dotnet global tool: amazon.lambda.tools v5.1.0
* OS Info: Windows 10 
* Build Environment: Terminal `dotnet` 
* Targeted .NET Platform:  netcoreapp3.1

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself

---

This is a :bug: bug-report

**Comment by SeanKilleen** [comment]
Note: I've checked the solution in #243. In that case, the namespace and assembly name seemed to be different. In my case, the assembly and the namespace share the same name, `LbsApi`.

**Comment by SeanKilleen** [comment]
Logs from `dotnet lambda package code.zip` from within the project directory:

```
Logs from `dotnet lambda package code.zip` from within the project directory:

```
Amazon Lambda Tools for .NET Core applications (5.1.0)
Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet

Executing publish command
Deleted previous publish folder
... invoking 'dotnet publish', working folder 'C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\bin\Release\netcoreapp3.1\publish'
... dotnet publish --output "C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\bin\Release\netcoreapp3.1\publish" --configuration "Release" --framework "netcoreapp3.1" /p:GenerateRuntimeConfigurationFiles=true --runtime linux-x64 --self-contained false 
... publish: Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
... publish: Copyright (C) Microsoft Corporation. All rights reserved.
... publish:   Determining projects to restore...
... publish:   All projects are up-to-date for restore.
... publish:   LbsApi -> C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\bin\Release\netcoreapp3.1\linux-x64\LbsApi.dll
... publish:   LbsApi -> C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\bin\Release\netcoreapp3.1\publish\
Zipping publish folder C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\bin\Release\netcoreapp3.1\publish to C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\code.zip
... zipping: Amazon.Lambda.APIGatewayEvents.dll
... zipping: Amazon.Lambda.ApplicationLoadBalancerEvents.dll
... zipping: Amazon.Lambda.AspNetCoreServer.dll
... zipping: Amazon.Lambda.Core.dll
... zipping: Amazon.Lambda.Logging.AspNetCore.dll
... zipping: Amazon.Lambda.Serialization.SystemTextJson.dll
... zipping: appsettings.Development.json
... zipping: appsettings.json
... zipping: aws-lambda-tools-defaults.json
... zipping: LbsApi
... zipping: LbsApi.deps.json
... zipping: LbsApi.dll
... zipping: LbsApi.pdb
... zipping: LbsApi.runtimeconfig.json
... zipping: web.config
Created publish archive (C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\code.zip).
Lambda project successfully packaged: C:\Users\skilleen\Repositories\[redacted]-api\src\LbsApi\code.zip
```

**Comment by SeanKilleen** [comment]
In reviewing some older threads elsewhere, e.g.  https://forums.aws.amazon.com/thread.jspa?threadID=285087, it sounds like this may be related to a permissions issue and not to the missing file. 

If anyone here knows of a way that I can diagnose/confirm this issue in my case, I'd greatly appreciate it. I'll begin digging into permissions in the interim.

**Comment by SeanKilleen** [comment]
This is looking like more of a red herring error related to permissions. 

If this proven to be the case, I'd be happy to work on improving the error handling if someone could point me to the repository that contains the logic that raises this error. Thanks!

**Comment by SeanKilleen** [comment]
As an attempt to remove permissions from the equation, I have granted `AWSLambda_FullAccess` and `AmazonS3FullAccess` to the role that my Lambda is using. The problem appears to persist (it is also unclear to me if the lambda deployment might be cached somewhere that I need to bust.)

**Comment by SeanKilleen** [comment]
@ashishdhingra I appreciate you looking into this. I discovered recently that this was my fault. I wrote a response to this thread but must not have posted it somehow.

The issue was indeed in my end, in terms of conceptualizing how Lambda deployment works in AWS. I discovered that the source code size for the lambda was listed as 140 bytes which raised some flags as you might imagine.

The misunderstanding on my part was that when deploying the lambda via Terraform, we'd had to set a bucket and a file location for the code. This led me to think that updating the file would be enough to trigger the deployment. However, it's completely reasonable that we'd need to specifically point the lambda to a new source file, even if that file is an updated file in the same location.

So, a tough lesson learned, but I appreciate your thorough response -- thank you! Closing this issue.
